src/Illuminate/Routing/Router.php

```diff
@@ -1228,4 +1230,20 @@ public function getPatterns()
     {
         return $this->patterns;
     }
+
+    /**
+     * Dynamically handle calls into the router instance.
+     *
+     * @param  string  $method
+     * @param  array  $parameters
+     * @return mixed
+     */
+    public function __call($method, $parameters)
+    {
+        if (static::hasMacro($method)) {
+            return $this->macroCall($method, $parameters);
+        }
+
+        return (new RouteRegistrar($this))->attribute($method, $parameters[0]);
```

Comment:
@rkgrep why???

The resource registrar has a lot of logic as to how it registers everything, so in some cases it makes sense to want to override it.

This route registrar serves the singular purpose of collection route configuration before the route is actually created. Doesn't make much sense to override it.